definitions/procedures/maintenance_mode/enable.rb

```diff
@@ -1,13 +0,0 @@
-module Procedures::MaintenanceMode
-  class Enable < ForemanMaintain::Procedure
```

Comment:
We should keep these procedures as they are already used in backup/restore. And it is indirectly used in upgrades.

Reply:
@mbacovsky,

For the upgrades, we are running steps using `:pre_migrations and :post_migrations` tags.
Is that ok if I added additional tags i.e `:maintenance_mode_on, :maintenance_mode_off` and find procedures using those tags in backup like 
[find_procedures(:maintenance_mode_on)](https://github.com/theforeman/foreman_maintain/pull/180/files#diff-a0ee368c0fb1cccf764aeade5e245985R133?).

In backup, it is already doing `Procedures::Service::Stop` and `Procedures::Service::Start` so I have not handled `crond` service part there separately.

Or you want me to move procedures from `MaintenanceModeStart and MaintenanceModeStop` scenarios into Procedures::MaintenanceMode::Enable and Procedures::MaintenanceMode::Disable?

Any suggestions?

Reply:
I think find_procedures could be acceptable solution.

Pros:
- it works out of the box
- it reads well

Cons:
- it is new concept. When composing new scenario you have to know there is no single procedure for turning on maint mode and you have look around how to do that
- it is hard to tell what is going on during 'maint mode on'. You have to search code for the tags

Other options:
- use Scenarios::MaintenanceMode::Start and let compose expand the steps. I'm not sure if we can nest scenarios. It would be easy to do.
- extract useful sets of steps in some module and re-use it later